lib/ansible/modules/cloud/amazon/rds_tag.py

```diff
@@ -0,0 +1,270 @@
+            botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't add tag to %s"
+                             % configured_tags['arn'])
+
+    response['message'] = 'Add %s tag to %s instance' % (
+        add_tags, configured_tags['arn'])
+
+    return response
+
+
+def remove_rds_tags(module, client, configured_tags, tags):
+    if not set(tags.keys()) <= set(configured_tags['tags'].keys()):
+        module.exit_json(message='nothing to tags.', changed=False)
+
+    if module.check_mode:
+        module.exit_json(message='check mode.', changed=True)
+
+    try:
+        response = client.remove_tags_from_resource(
+            ResourceName=configured_tags['arn'],
+            TagKeys=list(tags.keys()))
+    except (botocore.exceptions.ClientError,
+            botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e,
+                             msg="Couldn't remove tag to %s instance"
+                             % configured_tags['arn'])
+
+    response['message'] = 'Remove %s tag to %s instance' % (
+        tags.keys(), configured_tags['arn'])
+
+    return response
+
+
+def list_rds_arn(module, client, instance_name):
+    if instance_name is None:
+        try:
+            response_instances = client.describe_db_instances()
+        except (botocore.exceptions.ClientError,
+                botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, msg="Couldn't find RDS instance")
+    else:
+        try:
+            response_instances = client.describe_db_instances(
+                DBInstanceIdentifier=instance_name)
+        except (botocore.exceptions.ClientError,
+                botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, msg="Couldn't find %s instance"
+                                 % instance_name)
+
+    list_arn = [{'arn': instance['DBInstanceArn'],
+                 'identifier': instance['DBInstanceIdentifier']}
+                for instance in response_instances['DBInstances']]
+
+    return list_arn
+
+
+def list_rds_tags(module, client, list_arn, instance_name):
+    list_tags = []
+    for map_arn in list_arn:
+        try:
+            response_tags = client.list_tags_for_resource(
+                ResourceName=map_arn['arn'])
+        except (botocore.exceptions.ClientError,
+                botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, msg="Couldn't get %s instance's tag"
+                                 % instance_name)
+        tags = {}
+        for datum in response_tags['TagList']:
+            tags[datum['Key']] = datum['Value']
+
+        list_tags.append({'identifier': map_arn['identifier'],
+                          'tags': tags,
+                          'arn': map_arn['arn']})
+    return list_tags
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            instance_name=dict(type='str'),
```

Comment:
You can change this to `instance_name=dict(required=True)` and remove the check/failure below if it isn't provided. You don't need to specify `type='str'` because it is the default. I don't think the state 'list' belongs in the module, but for future reference, if you want something to be required for only certain states you could use:
```
module = AnsibleAWSModule(
    argument_spec=argument_spec,
    supports_check_mode=True,
    required_if=[('state', 'present', ('instance_name',)),
                 ('state', 'absent', ('instance_name',))]
)
```